jdaviz/core/marks.py

```diff
@@ -160,7 +160,11 @@ def __init__(self, viewer, x, **kwargs):
     def _update_reference_data(self, reference_data):
         if reference_data is None:
             return
-        self._update_unit(reference_data.get_object(cls=Spectrum1D).spectral_axis.unit)
+        try:
+            self._update_unit(reference_data.get_object(cls=Spectrum1D).spectral_axis.unit)
+        except (TypeError, ValueError):
+            # don't update x units for rampviz, which doesn't convert to Spectrum1D
```

Comment:
Would it be possible to have an explicit config check here rather than a try/except that could potentially hide real failures?

Reply:
I've changed this in the latest commit to check if `self.viewer.jdaviz_app.config == 'rampviz'`, as suggested.